SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -2679,7 +2680,7 @@ void CEulerSolver::SetTime_Step(CGeometry *geometry, CSolver **solver_container,
                                 unsigned short iMesh, unsigned long Iteration) {
 
   const bool viscous       = config->GetViscous();
-  const bool implicit      = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
+  const bool implicit      = (config->GetKind_TimeIntScheme() == EULER_IMPLICIT);
```

Comment:
What's the implication of switching away from the '_Flow()' time integration option? Shall we standardize on just one function for integration scheme? There's also the Turb() version. Whichever we choose we should make it very clear (this is an easy source for bugs).

Reply:
So far as I understand, the _Flow, _Turb versions are what is set in the config file and they are immutable (there is no "Set" method for them).
The mutable version (GetKind_TimeIntScheme) returns what is set by CConfig::SetGlobalParam, so maybe something like this was the intention all along?
The solver-specific versions are used in the constructors while other methods use the runtime version, that is what I changed in the compressible and incompressible solvers.
I also thought of passing a special value of iRKstep (akin to the NO_RK_STEP) (as I am not a big fan of relying on global state) to the pre-processing and residuals methods, and leave the boundary conditions un-touched. But this would not work for the discrete adjoint.
I don't think the chance for bugs is very high, SetGlobalParam is used only in Iteration::Iterate so unless the Integrations start calling each other there is little chance of mixing the settings.

Reply:
If you remove the _Flow() version can you also remove the _Turb() version over in the turb solvers with this PR, to avoid confusion? It will be simpler then if we always use only the single GetKind_TimeIntScheme() throughout. We might consider removing the _Flow() and _Turb() getters altogether and just set the correct IntScheme internally within the config class when switching btwn flow & turb, which would simplify further. In general, less options (w/ the same behavior) = less code = better.

In the future, we could apply a similar treatment for the ConvNumScheme, etc.

Reply:
It's done for that family of options where possible, constructors and so on need the specific versions for now, but I strongly agree with fewer code = better code.

Reply:
👍